GVFS/GVFS/CommandLine/SparseVerb.cs

```diff
@@ -235,10 +237,105 @@ protected override void Execute(GVFSEnlistment enlistment)
                     {
                         this.PruneFoldersOutsideSparse(tracer, enlistment, sparseTable);

```

Comment:
I think we need to have this return the the folders that are outside the sparse set to add to the list to refresh parents because if I have sparse set `A\B` and create a new folder `A\C` with a file, add and commit, then run `gvfs sparse --prune` or `gvfs sparse --set A\B --prune`, `A\C` should get pruned but it will not be in `foldersToAdd` or `foldersToRemove`.

Reply:
@kewillford great catch! I missed that we might prune folders that were never in the initial sparse set.

Reply:
@kewillford I confirmed that this scenario works correctly with my latest commit (and that it was broken before that change), thanks!